smartsim/_core/control/controller.py

```diff
@@ -591,3 +599,113 @@ def reload_saved_db(self, checkpoint_file):
             return orc
         finally:
             JM_LOCK.release()
+
+
+    def _enumerate_devices(self, dbobject: DBObject):
+        """Enumerate devices for a DBObject
+
+        :param dbobject: DBObject to enumerate
+        :type dbobject: DBObject
+        :return: list of device names
+        :rtype: list[str]
+        """
+        devices = []
+        if dbobject.device in ["CPU", "GPU"] and dbobject.devices_per_node > 1:
+            for device_num in range(dbobject.devices_per_node):
+                devices.append(f"{dbobject.device}:{str(device_num)}")
+        else:
+            devices = [dbobject.device]
+
+        return devices
+
+
+    def _set_ml_model(self, db_model, address, cluster=False):
```

Comment:
These two set methods could go into the ``_core/utils/redis.py`` file. Might clean up some of the imports and the base redis client is already imported there in case we want to to use that for validation or something in the future. 

I could also see these functions being used elsewhere. Possibly even with user facing functionality in the future. we should consider that here and split these into self contained functions.

Reply:
Done